### PR TITLE
Improve Chat Page

### PR DESCRIPTION
**Pull request by Theralun**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the chat page is pretty much unusable due to the incomprehensible chat list.
- Currently the message listeners are only present inside chats, but not on the chatPage

**What is the new behavior (if this is a feature change)?**
- Now the chats are automatically ordered by activity, greatly improving user experience.
- The message listeners now initialize with the chat Page to enable more functionality in the overview (unread messages, latest messages, ordered chat list)

### CODE REVIEW

File: src/components/pages/ChatPage/index.jsx [hunk 1]
```diff
@@ -64,10 +66,34 @@ class ChatPage extends React.Component {
...
   };
 
+  // Init on mount
   componentDidMount = () => {
     this.init();
   };
 
+  componentWillReceiveProps = (nextprops) => {
+    // If the chats don't exist right now but will exist in the next props,
+    // Start message handler for the chats
```
Comment (schettn) [inline src/components/pages/ChatPage/index.jsx:76]: Is this the correct way for multiline comments?

File: src/components/pages/ChatPage/index.jsx [hunk 2]
```diff
@@ -81,6 +107,29 @@ class ChatPage extends React.Component {
...
+      let co = messages[c.id];
+
+      if (co) {
+        order = [...order, { ...co[co.length - 1], chat: c }];
+      } else {
+        order = [...order, { mid: c.id, data: { sentTimestamp: 0 }, chat: c }];
+      }
+    });
+
+    //Put sorted array of messages into state
```
Comment (schettn) [inline src/components/pages/ChatPage/index.jsx:125]:
```suggestion
    // Put sorted array of messages into state
```

File: src/components/pages/ChatPage/index.jsx (hunk 1 above)
Comment (Aichnerc) [reply src/components/pages/ChatPage/index.jsx:77]:
```suggestion
 /*
  * If the chats don't exist right now but will exist in the next props,
  * start message handler for the chats
  */
```

Comment (Theralun) [reply src/components/pages/ChatPage/index.jsx]: Might have. did not pay attention. my bad

**Pull request merged**